heroku/heroku-buildpack-python
# Remove sqlite

Opt-in fix to SQLITE3 build errors on Heroku 16.

To opt in and use a Python version that was not built against SQLITE3, add the following config var to your application:

```
PYTHON_NO_SQLITE3=true
```

To test this on your app as a PR:
```
heroku buildpacks:set https://github.com/heroku/heroku-buildpack-python.git#remove-sqlite
```


Will Support:
Heroku 18
✅ Python 3.7.3
✅ Python 3.6.8
✅ Python 2.7.16

Heroku 16
✅ Python 3.7.3
✅ Python 3.6.8
✅ Python 2.7.16

Previously we explored the option to alter the sqlite dependency and vendoring flow - solution was workable but added maintenance burden. In addition, using SQLITE3 on Heroku is not advised due to the platform's ephemeral nature. For maintainability and to avoid breakage around an ideally unused system package that is not available in the base Heroku image on any current stack, I'm killin it ded.

Those who still wish to SQLITE3 despite being advised against using SQLITE3 on the platform may investigate using the [Apt buildpack](https://github.com/heroku/heroku-buildpack-apt), which is maintained by volunteers and does not come with official support from Heroku (but should work just fine).
